instrumentation/rediscala-1.8/javaagent/src/test/scala/io/opentelemetry/javaagent/instrumentation/rediscala/RediscalaClientTest.scala

```diff
@@ -0,0 +1,189 @@
+          .getConstructors()(0)
+          .newInstance(
+            host,
+            port,
+            Option.apply(null),
+            Option.apply(null),
+            "RedisClient",
+            Option.apply(null),
+            system,
+            RedisDispatcher("rediscala.rediscala-client-worker-dispatcher")
+          )
+          .asInstanceOf[RedisClient]
+    }
+  }
+
+  @AfterAll
+  def tearDown(): Unit = {
+    if (system != null) {
+      system.getClass.getMethod("terminate").invoke(system)
+    }
+    redisServer.stop()
+  }
+
+  @Test def testSetCommand(): Unit = {
+    val value = testing.runWithSpan(
+      "parent",
+      new ThrowingSupplier[Future[Boolean], Exception] {
+        override def get(): Future[Boolean] = {
+          redisClient.set("foo", "bar")
+        }
+      }
+    )
+
+    assertThat(Await.result(value, Duration.apply("3 second"))).isTrue
+    testing.waitAndAssertTraces(new Consumer[TraceAssert] {
+      override def accept(trace: TraceAssert): Unit =
+        trace.hasSpansSatisfyingExactly(
+          new Consumer[SpanDataAssert] {
+            override def accept(span: SpanDataAssert): Unit = {
+              span.hasName("parent").hasNoParent
+            }
+          },
+          new Consumer[SpanDataAssert] {
+            override def accept(span: SpanDataAssert): Unit = {
+              span
+                .hasName("SET")
+                .hasKind(CLIENT)
+                .hasParent(trace.getSpan(0))
+                .hasAttributesSatisfying(
+                  equalTo(DB_SYSTEM, REDIS),
+                  equalTo(SemconvStabilityUtil.maybeStable(DB_OPERATION), "SET")
+                )
+            }
+          }
+        )
+    })
+  }
+
+  @Test def testGetCommand(): Unit = {
+    val (write, value) = testing.runWithSpan(
+      "parent",
+      new ThrowingSupplier[
+        (Future[Boolean], Future[Option[String]]),
+        Exception
+      ] {
+        override def get(): (Future[Boolean], Future[Option[String]]) = {
+          val write = redisClient.set("bar", "baz")
+          val value = redisClient.get[String]("bar")
+          (write, value)
+        }
+      }
+    )
+
+    assertThat(Await.result(write, Duration.apply("3 second"))).isTrue
+    assertThat(
+      Await
+        .result(value, Duration.apply("3 second"))
+        .get
+    ).isEqualTo("baz")
+
+    testing.waitAndAssertTraces(new Consumer[TraceAssert] {
+      override def accept(trace: TraceAssert): Unit =
+        trace.hasSpansSatisfyingExactly(
+          new Consumer[SpanDataAssert] {
+            override def accept(span: SpanDataAssert): Unit = {
+              span.hasName("parent").hasNoParent
+            }
+          },
+          new Consumer[SpanDataAssert] {
+            override def accept(span: SpanDataAssert): Unit = {
+              span
+                .hasName("SET")
+                .hasKind(CLIENT)
+                .hasParent(trace.getSpan(0))
+                .hasAttributesSatisfying(
```

Comment:
is there a reason not to use `hasAttributesSatisfyingExactly` for these?

Reply:
Thanks, there isn't. Copy pasted it from the java conversion PR.